### PR TITLE
Better tab order

### PR DESCRIPTION
**Pull request by zainafzal08**

### CODE REVIEW

File: js/foam/ui/md/Halo.js [hunk 1]
```diff
@@ -151,11 +155,21 @@ CLASS({
...
+        } else {
+          this.r = Math.max(28, Math.max(this.$.clientWidth, this.parent.height));
+        }
         this.alpha = this.startAlpha;
-        var recentering = this.recentering;
+        const recentering = this.recentering;
         this.X.animate(this.easeInTime, function() {
-          if ( recentering ) {
+          if ( !this.animateGrowth ) {
+            // Do nothing.
```
Comment (E-T-) [inline js/foam/ui/md/Halo.js]: Why not

```
if (!this.animateGrowth) return;

if (recentering) {

} else {

}
```

File: js/foam/ui/md/Halo.js [hunk 2]
```diff
@@ -215,7 +229,27 @@ CLASS({
         if (this.focusType === 'mouse') return;
 
         this.focusType = 'keyboard';
+        // Save all of the halo properties.
+        const save = {
+          animateGrowth: this.animateGrowth,
+          startAlpha: this.startAlpha,
+          easeInTime: this.easeInTime,
+          easeOutTime: this.easeOutTime
```
Comment (E-T-) [inline js/foam/ui/md/Halo.js]: Add a trailing comma here.

Comment (E-T-) [reply js/foam/ui/md/Halo.js]: https://google.github.io/styleguide/jsguide.html#features-objects-use-trailing-comma

**Review by E-T- — COMMENTED** [review]
To double check --- is the idea here that you want to make fewer things tab selectable?

**Comment by zainafzal08** [comment]
> To double check --- is the idea here that you want to make fewer things tab selectable?

Yep! After a recent round of a11y feedback it was raised that we should have limited tab stops in which you can use the arrow keys to navigate.

File: js/foam/ui/md/Halo.js (hunk 1 above)
Comment (zainafzal08) [reply js/foam/ui/md/Halo.js]: We still need line 179 which sets the alpha.

File: js/foam/ui/md/Halo.js (hunk 2 above)
Comment (zainafzal08) [reply js/foam/ui/md/Halo.js]: Done, thank you!

File: js/foam/ui/md/Halo.js (hunk 1 above)
Comment (E-T-) [reply js/foam/ui/md/Halo.js]: Why not just move line 179 first and then do the if-else afterwards?

File: js/foam/graphics/AbstractCViewView.js [hunk 3]
```diff
@@ -157,7 +157,9 @@ CLASS({
     toHTML: function() { /* Creates the canvas element. */
       var className = this.className ? ' class="' + this.className + '"' : '';
       var title     = this.speechLabel ? ' aria-role="button" aria-label="' + this.speechLabel + '"' : '';
-      var tabIndex  = this.tabIndex ? ' tabindex="' + this.tabIndex + '"' : '';
+      // TabIndex can be '0' which evaluates to false so we directly check
+      // if it's undefined.
+      var tabIndex  = this.tabIndex !== undefined ? ' tabindex="' + this.tabIndex + '"' : '';
```
Comment (E-T-) [inline js/foam/graphics/AbstractCViewView.js]: Can this be `const` rather than `var`?

**Review by E-T- — APPROVED** [review]

File: js/foam/ui/md/Halo.js (hunk 1 above)
Comment (zainafzal08) [reply js/foam/ui/md/Halo.js]: Done!

File: js/foam/graphics/AbstractCViewView.js (hunk 3 above)
Comment (zainafzal08) [reply js/foam/graphics/AbstractCViewView.js]: Yep! Done!

**Pull request merged**